mne/viz/eyetracking/heatmap.py

```diff
@@ -0,0 +1,167 @@
+# Authors: Scott Huberty <seh33@uw.edu>
+#
+# License: BSD-3-Clause
+
+import numpy as np
+from scipy.ndimage import gaussian_filter
+
+
+from ..utils import plt_show
+from ...utils import _validate_type, logger, fill_doc
+
+
+@fill_doc
+def plot_gaze(
+    epochs,
+    width,
+    height,
+    sigma=25,
+    cmap=None,
+    alpha=None,
+    vmin=None,
+    vmax=None,
+    axes=None,
+    show=True,
+):
+    """Plot a heatmap of eyetracking gaze data.
+
+    Parameters
+    ----------
+    epochs : mne.Epochs
+        The :class:`~mne.Epochs` object containing eyegaze channels.
+    width : int
+        The width dimension of the plot canvas. For example, if the eyegaze data units
+        are pixels, and the participant screen resolution was 1920x1080, then the width
+        should be 1920.
+    height : int
+        The height dimension of the plot canvas. For example, if the eyegaze data units
+        are pixels, and the participant screen resolution was 1920x1080, then the height
+        should be 1080.
+    sigma : int | float
+        The amount of smoothing applied to the heatmap data. If ``None``,
+        no smoothing is applied. Default is 1.
+    cmap : matplotlib colormap | str | None
+        The :class:`~matplotlib.colors.Colormap` to use. Defaults to ``None``, meaning
+        the colormap will default to matplotlib's default.
+    alpha : int | float | None
+        The transparency value of the heatmap. If ``None``, the alpha value is set to 1,
+        meaning the heatmap colors are fully opaque. Default is ``None``.
+    vmin : float | None
+        The minimum value for the colormap. The unit is seconds, for dwell time
+        to the bin coordinate. If ``None``, the minimum value is set to the
+        minimum value of the heatmap. Default is ``None``.
+    vmax : float | None
+        The maximum value for the colormap. The unit is seconds, for dwell time
+        to the bin coordinate. If ``None``, the maximum value is set to the
+        maximum value of the heatmap. Default is ``None``.
+    axes : matplotlib.axes.Axes | None
+        The axes to plot on. If ``None``, a new figure and axes are created.
+        Default is ``None``.
+    show : bool
+        Whether to show the plot. Default is ``True``.
+
+    Returns
+    -------
+    fig : instance of matplotlib.figure.Figure
+        The resulting figure object for the heatmap plot.
+
+    Notes
+    -----
+    .. versionadded:: 1.6
+    """
+    from mne import BaseEpochs
+    from mne._fiff.pick import _picks_to_idx
+
+    _validate_type(epochs, BaseEpochs, "epochs")
```

Comment:
Should use `width = _ensure_int(width, 'width')` and same for height, also `_validate_type(..., float, ...)` on `alpha` and `sigma`

Reply:
Also , just to be explicit, I'd use

```
 _validate_type(alpha, (int, float, None), "alpha")
 _validate_type(sigma, (int, float, None), "sigma")
```

Because if someone passes `alpha=1`, instead of `1.0`, or `sigma=25` (not `25.0`), the function should still just work?

Reply:
you can use `_validate_type(alpha, "numeric", "alpha)` as a shorthand for "anything numeric", we often use this for float/int. I guess you could allow width/height to be float as well

Reply:
Will keep the `_ensure_int` on width/height for now (to revisit once a `mne.preprocessing.eyetrackingconvert_units` function is hopefully implemented).